supervisor/shared/serial.c

```diff
@@ -207,10 +207,17 @@ char serial_read(void) {
 
     #if CIRCUITPY_WEB_WORKFLOW
     if (websocket_available()) {
-        return websocket_read_char();
+        char c = websocket_read_char();
```

Comment:
getc() can differentiate between EOF and one of the valid values of 'char' by having the return type of "int". However, `websocket_read_char()` returns `char`.  One manpage explains how (f)getc works as follows:

> fgetc()  reads  the next character from stream and returns it as an *unsigned char* cast to an int, or EOF on end of file or error.